include/bm/bm_sim/actions.h

```diff
@@ -636,6 +636,7 @@ class ActionFn :  public NamedP4Object {
   void parameter_push_back_string(const std::string &str);
 
   void push_back_primitive(ActionPrimitive_ *primitive);
+  void push_back_source_info(std::unique_ptr<SourceInfo> source_info);
```

Comment:
I would make it an optional parameter to `push_back_primitive` instead, it makes more sense IMO